File: server/bcephal-base/src/main/java/com/moriset/bcephal/domain/EditorDataFilter.java
/**
 * 
 */
package com.moriset.bcephal.domain;

import com.moriset.bcephal.domain.filters.ColumnFilter;

import lombok.Data;

/**
 * @author Joseph Wambo
 *
 */
@Data
public class EditorDataFilter {
	
	private DataSourceType dataSourceType;	
	private Long dataSourceId;
	private Long id;
	private Long secondId;
	private String subjectType;
	private boolean showAllMeasureTypes;
	private boolean showPeriodWithIntervalls;
	private boolean showModelWithAttributes;
	private boolean showModelWithTargetCustom;
	private boolean newData;

	
	private ColumnFilter ColumnFilters;
	
	public EditorDataFilter() {
		this.showAllMeasureTypes = false;
		this.showPeriodWithIntervalls = false;
		this.showModelWithAttributes = true;
		this.showModelWithTargetCustom = true;
		this.newData = false;
		this.dataSourceType = DataSourceType.UNIVERSE;
	}
	
	public EditorDataFilter(Long id) {
		this.id = id;
		this.showAllMeasureTypes = false;
		this.showPeriodWithIntervalls = false;
		this.showModelWithAttributes = true;
		this.showModelWithTargetCustom = true;
		this.newData = false;
		this.dataSourceType = DataSourceType.UNIVERSE;
	}

}

File: client/bcephal-client-model/Base/EditorDataFilter.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Bcephal.Models.Base
{
    public class EditorDataFilter
    {
        public long? Id { get; set; }

        public long? SecondId { get; set; }

        public string SubjectType { get; set; }
        public bool ShowAllMeasureTypes { get; set; }
        public bool ShowPeriodWithIntervalls { get; set; }
        public bool ShowModelWithAttributes { get; set; }
        public bool ShowModelWithTargetCustom { get; set; }

        public bool NewData { get; set; }

        public EditorDataFilter()
        {
            //this.projectCode = ApplicationManager.Instance.Project.code;
            this.ShowAllMeasureTypes = false;
            this.ShowPeriodWithIntervalls = false;
            this.ShowModelWithAttributes = true;
            this.ShowModelWithTargetCustom = true;
            this.NewData = false;
        }

        public EditorDataFilter(long? id, string subjectType, bool newData = false) : this()
        {
            this.Id = id;
            this.SubjectType = subjectType;
            this.NewData = newData;
        }

    }
}
